### ISSUE TITLE
Validation of compatible ibek / ibek-support / ioc-XXX

### ISSUE THREAD

**Issue by gilesknap** [epics-containers/ibek#133]
The above 3 things need to be in sync for a Generic IOC CI build to succeed.

- ibek-support uses ibek's schema for Support YAML and also its CLI in the install.sh files.- 
- ioc-XXX Dockerfile pip installs a version of ibek and tracks a commit hash of the submodule ibek-support. 
- The Dockerfile needs to find all of the support modules it expects in ibek-support. 

It is very easy to get this all working locally and then forget to make sure all the same things are available to the CI.

@GDYendell and I discussed this and thought it might be possible to make a useful pre-commit that could try to verify this for us. It would need to:

- make sure the locally installed ibek and the pip install in Dockerfile match
- make sure the submodule is committed and pushed
- have some way of knowing whether the submodule is compatible with the current ibek version
- anything else?

@coretl interested in your views on this.

**Comment by coretl** [comment]
This might be useful, although hopefully ibek will stabilise, so then maybe a `ibek check-version-exceeds 0.2.1` in each `install.sh` would be better than a strict equality check...

Checking submodule is committed and pushed is something that could be useful, I guess you would write this as a [pre-commit](https://pre-commit.com/) hook?

**Comment by gilesknap** [comment]
Yes re precomit hook - I like the check-version-exceeds thing (we are making breaking changes right now that would not work with this but that should settle down).

It would be nice if precomit could scan the ibek-support to verify check-version-exceeds somehow though - in which case maybe its something outside of the install.sh files.

**Comment by gilesknap** [comment]
I believe this has been resolved with the new version scheme and update flow diagram that I will be including in this weeks version of the documentation.
